bundle/config/mutator/run_as_test.go

```diff
@@ -0,0 +1,46 @@
+package mutator
+
+import (
+	"slices"
+	"testing"
+
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/convert"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+// Every new resource type added to DABs should either be classified in the allow
+// list or the deny list for run_as other support.
+func TestAllResourceTypesAreClassifiedForRunAs(t *testing.T) {
```

Comment:
Originally we should have had unit test coverage because of the e2e tests in the `bundle/tests` directory. Not sure why that did not get picked up.

Definitely have the coverage though, now specially since we use the typed configuration everywhere.